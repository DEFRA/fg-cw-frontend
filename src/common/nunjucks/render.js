import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
