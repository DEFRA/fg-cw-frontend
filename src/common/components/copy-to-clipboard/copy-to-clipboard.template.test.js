import { describe, expect, test } from "vitest";
import { render } from "../../nunjucks/render.js";

describe("copy-to-clipboard template", () => {
  test("renders with default parameters", () => {
    const component = render("copy-to-clipboard", {});
    expect(component).toMatchSnapshot();
  });

  test("renders with custom text to copy", () => {
    const component = render("copy-to-clipboard", {
      text: "user@example.com",
    });
    expect(component).toMatchSnapshot();
  });

  test("renders with custom button text", () => {
    const component = render("copy-to-clipboard", {
      text: "secret-key-123",
      buttonText: "Copy Key",
    });
    expect(component).toMatchSnapshot();
  });

  test("renders with custom feedback text", () => {
    const component = render("copy-to-clipboard", {
      text: "some-value",
      feedbackText: "Value copied!",
    });
    expect(component).toMatchSnapshot();
  });

  test("renders with custom CSS classes", () => {
    const component = render("copy-to-clipboard", {
      text: "test-value",
      classes: "custom-class another-class",
    });
    expect(component).toMatchSnapshot();
  });

  test("renders with all custom parameters", () => {
    const component = render("copy-to-clipboard", {
      text: "full-example@test.com",
      buttonText: "Copy Email",
      feedbackText: "Email copied successfully!",
      classes: "govuk-button govuk-button--secondary",
    });
    expect(component).toMatchSnapshot();
  });
});
