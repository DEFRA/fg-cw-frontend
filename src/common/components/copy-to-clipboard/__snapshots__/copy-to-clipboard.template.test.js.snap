// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`copy-to-clipboard template > renders with all custom parameters 1`] = `
"
<div
  class="copy-to-clipboard-container"
  style="position: relative; display: inline-block;"
>
  <button
    type="button"
    class="copy-to-clipboard-button govuk-button govuk-button--secondary"
    data-module="copy-to-clipboard"
    data-text="full-example@test.com"
    aria-label="Copy full-example@test.com to clipboard"
  >
    Copy Email
  </button>
  <span class="govuk-tag copy-to-clipboard-feedback" style="display: none;">
    Email copied successfully!
  </span>
</div>
"
`;

exports[`copy-to-clipboard template > renders with custom CSS classes 1`] = `
"
<div
  class="copy-to-clipboard-container"
  style="position: relative; display: inline-block;"
>
  <button
    type="button"
    class="copy-to-clipboard-button custom-class another-class"
    data-module="copy-to-clipboard"
    data-text="test-value"
    aria-label="Copy test-value to clipboard"
  >
    Copy
  </button>
  <span class="govuk-tag copy-to-clipboard-feedback" style="display: none;">
    Copied to clipboard
  </span>
</div>
"
`;

exports[`copy-to-clipboard template > renders with custom button text 1`] = `
"
<div
  class="copy-to-clipboard-container"
  style="position: relative; display: inline-block;"
>
  <button
    type="button"
    class="copy-to-clipboard-button "
    data-module="copy-to-clipboard"
    data-text="secret-key-123"
    aria-label="Copy secret-key-123 to clipboard"
  >
    Copy Key
  </button>
  <span class="govuk-tag copy-to-clipboard-feedback" style="display: none;">
    Copied to clipboard
  </span>
</div>
"
`;

exports[`copy-to-clipboard template > renders with custom feedback text 1`] = `
"
<div
  class="copy-to-clipboard-container"
  style="position: relative; display: inline-block;"
>
  <button
    type="button"
    class="copy-to-clipboard-button "
    data-module="copy-to-clipboard"
    data-text="some-value"
    aria-label="Copy some-value to clipboard"
  >
    Copy
  </button>
  <span class="govuk-tag copy-to-clipboard-feedback" style="display: none;">
    Value copied!
  </span>
</div>
"
`;

exports[`copy-to-clipboard template > renders with custom text to copy 1`] = `
"
<div
  class="copy-to-clipboard-container"
  style="position: relative; display: inline-block;"
>
  <button
    type="button"
    class="copy-to-clipboard-button "
    data-module="copy-to-clipboard"
    data-text="user@example.com"
    aria-label="Copy user@example.com to clipboard"
  >
    Copy
  </button>
  <span class="govuk-tag copy-to-clipboard-feedback" style="display: none;">
    Copied to clipboard
  </span>
</div>
"
`;

exports[`copy-to-clipboard template > renders with default parameters 1`] = `
"
<div
  class="copy-to-clipboard-container"
  style="position: relative; display: inline-block;"
>
  <button
    type="button"
    class="copy-to-clipboard-button "
    data-module="copy-to-clipboard"
    data-text="Copy"
    aria-label="Copy Copy to clipboard"
  >
    Copy
  </button>
  <span class="govuk-tag copy-to-clipboard-feedback" style="display: none;">
    Copied to clipboard
  </span>
</div>
"
`;
